servicelayer/archive/s3.py

```diff
@@ -99,8 +101,11 @@ def load_file(self, content_hash, file_name=None, temp_path=None):
             self.client.download_file(self.bucket, key, str(path))
             return path
 
-    def generate_url(self, content_hash, file_name=None, mime_type=None):
-        key = self._locate_key(content_hash)
+    def generate_url(self, content_hash=None, file_name=None, mime_type=None, key=None):  # noqa
```

Comment:
I wonder if there should be two separate functions here: `generate_url` is for temporary access to an unnamed resource, while some other method allows you to publish a name-based resource for a much longer period, potentially indefinitely?

Reply:
Hmm makes sense. We can use the same API to publish data dumps too